modules/turbsim/src/TSsubs.f90

```diff
@@ -195,41 +156,18 @@ SUBROUTINE CalcFourierCoeffs_IEC( p, U, PhaseAngles, S, V, TRH_in, ErrStat, ErrM
          !   use H matrix to calculate coefficients
          ! -----------------------------------------------
          
-         CALL Coh2H(    p, IVec, IFreq, TRH, S, ErrStat2, ErrMsg2 )
-         !$OMP CRITICAL 
-         CALL SetErrStat(ErrStat2, ErrMsg2, ErrStat, ErrMsg, 'CalcFourierCoeffs_IEC')
-         !$OMP END CRITICAL
+         CALL Coh2H(p, IVec, IFreq, TRH, S, ErrStat2, ErrMsg2)
          if (ErrStat2 >= AbortErrLev) then
-            OMPerr = .true.
+            !$OMP CRITICAL 
```

Comment:
Nice!  that's a good way to handle it -- reduces the number OMP CRITICAL usages dramatically